src/ArrayInterface.jl

```diff
@@ -106,6 +106,19 @@ has_sparsestruct(x::Type{<:Bidiagonal}) = true
 has_sparsestruct(x::Type{<:Tridiagonal}) = true
 has_sparsestruct(x::Type{<:SymTridiagonal}) = true
 
+"""
+    issingular(A::AbstractMatrix)
+
+Determine whether a given abstract matrix is singular.
+"""
+issingular(A::Matrix) = !issuccess(lu(A, check=false))
+issingular(A::UniformScaling) = A.λ == 0
+issingular(A::Diagonal) = any(iszero,A.diag)
```

Comment:
One could add here
```julia
issingular(B::Bidiagonal) = any(iszero, A.dv)
issingular(S::SymTridiagonal) = diaganyzero(iszero, ldlt(S).data)
issingular(T::Tridiagonal) = !issuccess(lu(A, check=false))
```
And perhaps as a fallback (which fulfills the promise of the docstring)
```julia
issingular(A::AbstractMatrix) = issingular(Matrix(A))
issingular(A::AbstractSparseMatrix) = !issuccess(lu(A, check=false))
```